docs/WinRTAPIContracts.md

```diff
@@ -0,0 +1,76 @@
+# Windows App SDK API Contracts
+
+## Summary
+Customers trust Microsoft that the products we make are sound, and will work.  Windows App SDK (WASDK) has recently released version 1.0.  As WASDK evolvs and more developers use WASDK in their own projects WASDK needs a way to guarantee that the API’s that ship are stable and that API versions are well documented.
+To keep customers’ trust in WASDK, and to make sure the WASDK API’s are
+1.  Binary compatible
+2.  Stable
+3.  Customizable so developers can include only the parts they want
+
+Contracts, and contract versions should be added to WASDK.
+
+## Architecture Decisions
+1. Contract boundaries
+Contracts will be split among the winmd boundaries.  This is a good medium between having 1 contract for everything, and 1 contract per class.
+2. Contract numbering.
+Contracts will be numbered with MAJOR.MINOR.  Any update to the API is a MAJOR change.  Any servicing update is a MINOR change.
+3. Enforcement
+The build systems will have enforcement gates to make sure that contracts are present, up to date, and there are no conflicts
+
+## Example
+    namespace Microsoft.Windows.System
+    {
+        [contractversion(2.0)]
+        apicontract MicrosoftWindowsSystem
+        
+        // version 2.0
+        [Contract(MicrosoftWindowsSystem(2.0)]
+        public enum Scope
+        {
+            user,
+            machine
+        }
+
+        [Contract(MicrosoftWindowsSystem(2.0)]
+        public class EnvironmentManager
+        {
+            public String GetEnvironmentVariable(String name, Scope scope);
+        }
+        
+        // servicing
+        [Contract(MicrosoftWindowsSystem(1.5)]        
+        public class EnvironmentManager
+        {
+            public String GetEnvironmentVariable(String name, String scope);
+        }
+
+        // base version
+        [contract(MicrosoftWindowsSystem, 1.0),]
+        public class EnvironmentManager
+        {
+            public String GetEnvironmentVariable(String name);
+        }
+    }
+
+## 1.1 ship blocker
+Any API's that changed between 1.0 and 1.1 need to add a contract or 1.1 will not ship.
+
+## Telemetry
+Because all API's have a contract version data can be used to see
+ 1. How many people are using each API
```

Comment:
I'm not seeing how API contract versions propogate through telemetry. If we add telemetry to include component=`Microsoft.UI.XAML.Control.Button` AND apicontract=`1.1` but I expect winappsdkversion=`1.1.8` (or whatever) is the more useful superset of information (given version you can determine contracts, but not vice versa).

Please elaborate how contracts can be used to via telemetry to provide richer information than available via the alternatives (like winappsdk version).

Reply:
That's a good point about the 1-n mapping between winappsdk and api contracts. Capturing the winappsdk version for telemetry is the more valuable data point.

Reply:
Oooohhh.  Yeah.  I think I was missing a step in my logic.  I thought developers could choose what versions of APIs to use.  If that isn't the case and developers get whatever version of the API is in their version of Windows App SDK then it does not make sense to include telemetry for API versions.